ui/src/pages/publications/[id]/index.tsx

```diff
@@ -65,13 +65,14 @@ export const getServerSideProps: Types.GetServerSideProps = async (context) => {
     if (token) {
         //check if user is author / co author. if so the bookmark icon won't display
         const currentUser = jwt.decode(token) as any;
-
-        if (currentUser.id) {
-            const isCoAuthor = publication?.coAuthors.some((author) => author.id == currentUser.id);
-            const isOwner = currentUser.id === publication.createdBy;
-
-            if (!isCoAuthor && !isOwner) {
-                isBookmarkVisible = true;
+        if (Date.now() <= currentUser.exp * 1000) {
+            if (currentUser.id) {
+                const isCoAuthor = publication?.coAuthors.some((author) => author.id == currentUser.id);
+                const isOwner = currentUser.id === publication.createdBy;
+
+                if (!isCoAuthor && !isOwner) {
```

Comment:
We can leave this as it is for now, but we will need to move this logic on the client side, into the page and check the user directly(instead of decoding the jwt) once we fix the expired jwt issue.